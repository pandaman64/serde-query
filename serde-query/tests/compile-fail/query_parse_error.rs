#[derive(serde_query::Deserialize)]
struct A {
    #[query(r#"."#)]
    missing_field: String,
    #[query(r#".[kubernetes_clusters]"#)]
    field_in_bracket: String,
    #[query(r#".ðŸ˜Ž"#)]
    unsupported_char: String,
}

fn main() {}