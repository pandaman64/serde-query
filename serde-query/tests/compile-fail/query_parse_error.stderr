error: 1..1: expected '[' or an identifier, got EOF
 --> $DIR/query_parse_error.rs:3:5
  |
3 |     #[query(r#"."#)]
  |     ^^^^^^^^^^^^^^^^

error: 1..22: expected an index or a quoted field inside indexing, got Field
 --> $DIR/query_parse_error.rs:5:5
  |
5 |     #[query(r#".[kubernetes_clusters]"#)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: 1..5: expected '[' or an identifier, got Error
 --> $DIR/query_parse_error.rs:7:5
  |
7 |     #[query(r#".ðŸ˜Ž"#)]
  |     ^^^^^^^^^^^^^^^^^^
